crates/matrix-sdk/src/room/joined.rs

```diff
@@ -1025,3 +1016,56 @@ impl Joined {
         self.client.send(request, None).await
     }
 }
+
+/// Receipts to send all at once.
+#[derive(Debug, Clone, Default)]
+pub struct Receipts {
+    fully_read: Option<OwnedEventId>,
+    read_receipt: Option<OwnedEventId>,
+    private_read_receipt: Option<OwnedEventId>,
+}
+
+impl Receipts {
+    /// Create an empty `Receipts`.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Set the last event the user has read.
+    ///
+    /// It means that the user has read all the events before this event.
+    ///
+    /// This is a private marker only visible by the user.
+    ///
+    /// Note that this is technically not a receipt as it is persisted in the
+    /// room account data.
+    pub fn fully_read_marker(mut self, event_id: impl Into<OwnedEventId>) -> Self {
```

Comment:
Ah sorry, what I really meant was `Into<Option<OwnedEventId>>` so an if-let like in the FFI code is not necessary for passing one of these event IDs optionally.